### PR TITLE
fixed issue in retriving licence agreement

### PR DESCRIPTION
**Pull request by hasuniea**

### CODE REVIEW

File: modules/apps/jaggery/emm/src/emm/modules/device.js [hunk 1]
```diff
@@ -319,16 +319,19 @@ deviceModule = function () {
     publicMethods.getLicense = function (deviceType, languageCode) {
         var url;
         var license;
-        if(deviceType != null && deviceType != undefined && deviceType == "windows"){
+        if (deviceType != null && deviceType != undefined && deviceType == "windows") {
```
Comment (DilanUA) [inline modules/apps/jaggery/emm/src/emm/modules/device.js]: you can simply use - if (deviceType == "windows"), and use a constant for accessing deviceType

File: modules/apps/jaggery/emm/src/emm/modules/device.js [hunk 2]
```diff
@@ -319,16 +319,19 @@ deviceModule = function () {
...
             url = mdmProps["httpURL"] + "/mdm-windows-agent/services/device/license";
-        } else{
+        } else if (deviceType != null && deviceType != undefined && deviceType == "ios") {
```
Comment (DilanUA) [inline modules/apps/jaggery/emm/src/emm/modules/device.js]: you can simply use - if (deviceType == "ios"), and use a constant for accessing deviceType

File: modules/apps/jaggery/emm/src/emm/modules/device.js [hunk 3]
```diff
@@ -319,16 +319,19 @@ deviceModule = function () {
...
             url = mdmProps["httpsURL"] + "/ios-enrollment/license/";
         }
-        serviceInvokers.XMLHttp.get(url, function (responsePayload) {
-            license = responsePayload.text;
-        }, function (responsePayload) {
-            return null;
-        });
+
+        if (url != null && url != undefined) {
```
Comment (DilanUA) [inline modules/apps/jaggery/emm/src/emm/modules/device.js]: simply use if (url) here. that will check these all.

File: modules/apps/jaggery/emm/src/emm/modules/device.js [hunk 4]
```diff
@@ -319,16 +319,19 @@ deviceModule = function () {
...
+            serviceInvokers.XMLHttp.get(url, function (responsePayload) {
+                license = responsePayload.text;
+            }, function (responsePayload) {
+                return null;
+            });
+        }
```
Comment (DilanUA) [inline modules/apps/jaggery/emm/src/emm/modules/device.js]: it's useless here to have a error callback method just to return null.

File: modules/apps/jaggery/emm/src/emm/modules/device.js (hunk 1 above)
Comment (hasuniea) [inline modules/apps/jaggery/emm/src/emm/modules/device.js]: fixed the issue from https://github.com/wso2/product-mdm/pull/156

File: modules/apps/jaggery/emm/src/emm/modules/device.js (hunk 2 above)
Comment (hasuniea) [inline modules/apps/jaggery/emm/src/emm/modules/device.js]: fixed the issue from #156

File: modules/apps/jaggery/emm/src/emm/modules/device.js (hunk 4 above)
Comment (hasuniea) [inline modules/apps/jaggery/emm/src/emm/modules/device.js]: fixed it from https://github.com/wso2/product-mdm/pull/156

File: modules/apps/jaggery/emm/src/emm/modules/device.js (hunk 3 above)
Comment (hasuniea) [inline modules/apps/jaggery/emm/src/emm/modules/device.js]: there is a extra call from the device browser,therefore need to check that url.

**Pull request merged**